ngraph/core/reference/include/ngraph/runtime/reference/einsum.hpp

```diff
@@ -0,0 +1,1210 @@
+                {
+                    // transpose is not needed if common labels, reduced labels
+                    // and separate labels indices go concurrently
+                    std::vector<int64_t> labels_inds = common_labels_inds;
+                    labels_inds.insert(
+                        labels_inds.end(), reduced_labels_inds.begin(), reduced_labels_inds.end());
+                    labels_inds.insert(labels_inds.end(),
+                                       separate_labels_inds.begin(),
+                                       separate_labels_inds.end());
+                    if (is_range_0_to_n(labels_inds))
+                    {
+                        is_separate_first = false;
+                        return input_subscript;
+                    }
+
+                    // transpose is not needed if common labels, separate labels
+                    // and reduced labels indices go concurrently
+                    labels_inds = common_labels_inds;
+                    labels_inds.insert(labels_inds.end(),
+                                       separate_labels_inds.begin(),
+                                       separate_labels_inds.end());
+                    labels_inds.insert(
+                        labels_inds.end(), reduced_labels_inds.begin(), reduced_labels_inds.end());
+                    if (is_range_0_to_n(labels_inds))
+                    {
+                        is_separate_first = true;
+                        return input_subscript;
+                    }
+
+                    auto labels = ngraph::opset7::Einsum::extract_labels(input_subscript);
+                    std::string required_subscript = "";
+                    for (auto index : labels_inds)
+                    {
+                        required_subscript += labels[index];
+                    }
+                    is_separate_first = true;
+                    return required_subscript;
+                }
+
+                std::unordered_map<std::string, std::vector<size_t>>
+                    compute_label_dim_map(const Rank& input_rank,
+                                          const std::string& input_subscript)
+                {
+                    const std::string ellipsis = "...";
+                    auto labels = ngraph::opset7::Einsum::extract_labels(input_subscript);
+                    size_t input_rank_length = labels.size();
+                    if (input_rank.is_dynamic() &&
+                        std::find(labels.begin(), labels.end(), ellipsis) != labels.end())
+                    {
+                        NGRAPH_CHECK(
+                            false,
+                            "Input rank cannot be dynamic in case of ellipsis in input subscript");
+                    }
+                    else if (input_rank.is_static())
+                    {
+                        input_rank_length = input_rank.get_length();
+                    }
```

Comment:
```suggestion
                    NGRAPH_CHECK(
                            input_rank.is_static() || std::find(labels.begin(), labels.end(), ellipsis) == labels.end(),
                            "Input rank cannot be dynamic in case of ellipsis in input subscript");
                   if (input_rank.is_static())
                    {
                        input_rank_length = input_rank.get_length();
                    }
```

Reply:
Done.